mono/mini/interp/transform.c

```diff
@@ -1584,7 +1584,8 @@ interp_transform_call (TransformData *td, MonoMethod *method, MonoMethod *target
 		if (m_class_is_valuetype (constrained_class) && (target_method->klass == mono_defaults.object_class || target_method->klass == m_class_get_parent (mono_defaults.enum_class) || target_method->klass == mono_defaults.enum_class)) {
 			if (target_method->klass == mono_defaults.enum_class && (td->sp - csignature->param_count - 1)->type == STACK_TYPE_MP) {
 				/* managed pointer on the stack, we need to deref that puppy */
-				ADD_CODE (td, MINT_LDIND_I);
+				/* Always load the entire stackval, to handle also the case where the enum has long storage */
+				ADD_CODE (td, MINT_LDIND_I8);
```

Comment:
heads-up: this is going to conflict with https://github.com/mono/mono/pull/13154